src/KingSystem/System/QuestMgr.h

```diff
@@ -0,0 +1,32 @@
+#pragma once
+
+#include <container/seadPtrArray.h>
+#include <heap/seadDisposer.h>
+#include <prim/seadSafeString.h>
+#include "KingSystem/System/Quest.h"
+#include "agl/Utils/aglParameter.h"
+
+namespace ksys {
+
+class QuestMgr {
+    SEAD_SINGLETON_DISPOSER(QuestMgr)
+public:
+    QuestMgr();
+    ~QuestMgr();
+
+    bool setQuestStep(const sead::SafeString& quest_name, const sead::SafeString& step_name,
+                      bool copy_name, bool force_run_telop, bool setAocVersionFlag1);
+
+    struct Quests : public sead::PtrArrayImpl {};
```

Comment:
Are you sure it inherits sead::PtrArrayImpl? afaik PtrArrayImpl is never used directly; `sead::PtrArray<T>` is typically used instead.

Also, can this be a simple type alias (i.e. `using Quests = sead::PtrArray<Quest>;`? (You can forward declare Quest with most pointer-based sead containers.)